lib/web-server.js

```diff
@@ -61,7 +63,14 @@ var createWebServer = function (injector, emitter) {
       common.serve404(response, request.url)
     })
 
-  var server = http.createServer(handler)
+  var serverClass = http
+  var serverArguments = [handler]
+
+  if (config.protocol === 'https:') {
+    serverClass = https
+    serverArguments.unshift(config.httpsServerOptions || {})
+  }
+  var server = serverClass.createServer.apply(null, serverArguments)
```

Comment:
Could you please add a test for this, making sure when `https` is set, it's actually used